### DISCUSSION TITLE
Relationship boilerplate

### DISCUSSION THREAD

**Discussion by stereobooster** [adonisjs/core#4030] [Help]
I'm new to Adonis and maybe doesn't understand something. For every `hasOne` relationship I have to declare keys. Can't Adonis figure out column based on relationship itself?

```ts
  @hasOne(() => Node, {
    localKey: 'to_node_id',
    foreignKey: 'id',
  })
  public to: HasOne<typeof Node>

  @column({ serializeAs: null })
  public to_node_id: number
```

So I would not need to specify `public to_node_id: number`?

**Comment by Barbapapazes** [comment]
Hello, can you explain a little more your issue?

**Comment by WeyIin** [comment]
Can you show us your migration? 

But basically, if you name columns appropriately, it will know what and how to link and all you'd have to do is:

```javascript
  @hasOne(() => Node)
  public to: HasOne<typeof Node>
```
